### PR TITLE
Feature/delete remedy

### PR DESCRIPTION
**Pull request by AbdulrahmanTabbakh**

### CODE REVIEW

**Review by JessicaLoers — CHANGES_REQUESTED** [review]
**Great work on the project so far! ✨** 

I have a few suggestions that could enhance the clarity, reusability, and maintainability of your code. 

Furthermore consider modify and renaming the  component  `ModalDelete.js` to something more generic, like `ConfirmationDialog`. To make the component more reusable across different parts of your application, as it won’t be tied to a specific context (like deleting a remedy).  **_This could also be a task for refactoring in the future._**

File: components/ModalDelete.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    setShowModal(true);
+  };
+  const handleCloseModal = () => {
+    setShowModal(false);
+  };
+
+  const handleConfirmDelete = () => {
+    setShowModal(false);
+    onDelete();
+  };
```
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:17]: function handleConfirmDelete() {
  setShowModal(false);
  onDelete();
}

File: components/ModalDelete.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+  return (
+    <>
+      <DeleteButton onClick={handleDeleteClick}>Delete Remedy</DeleteButton>
+
+      {showModal && (
+        <Modal onClick={handleCloseModal}>
+          <ModalContent onClick={(e) => e.stopPropagation()}>
+            <CloseButton onClick={handleCloseModal}>&times;</CloseButton>
```
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:26]: The stopPropagation function is prevents the click event on the modal content from triggering the click event on the modal's background, which would close the modal. However, we can throw it out here if considered : 

{showModal && (
  <Modal onClick={handleCloseModal}>
    <ModalContent></ModalContent>
     </Modal>
     )}

File: components/ModalDelete.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    onDelete();
+  };
+
+  return (
+    <>
+      <DeleteButton onClick={handleDeleteClick}>Delete Remedy</DeleteButton>
+
+      {showModal && (
+        <Modal onClick={handleCloseModal}>
+          <ModalContent onClick={(e) => e.stopPropagation()}>
```
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:25]: The stopPropagation function is prevents the click event on the modal content from triggering the click event on the modal's background, which would close the modal. However, we can throw it out here if considered : 


```
{showModal && (
  <Modal onClick={handleCloseModal}>
    <ModalContent></ModalContent>
     </Modal>
     )}
```

File: components/ModalDelete.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:26]: Updated code according to considerations

```
{showModal && (
  <Modal>
    <ModalContent>
      <button onClick={handleConfirmDelete}>Delete</button>
      <button onClick={toggleModal}>Cancel</button>
    </ModalContent>
  </Modal>
)}
```

File: components/ModalDelete.js [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            <CloseButton onClick={handleCloseModal}>&times;</CloseButton>
+            <Container>
+              <h1>Delete Remedy</h1>
```
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:28]: <Container>
              <h2>Delete Remedy</h2>

File: components/ModalDelete.js [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+`;
+
+const CloseButton = styled.span`
+  position: absolute;
+  right: 15px;
+  top: 10px;
+  font-size: 40px;
+  font-weight: bold;
+  color: #f8fbca;
+  cursor: pointer;
```
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:94]: deletion of L. 94

File: pages/remedy/[id].js [hunk 6]
```diff
@@ -12,9 +14,13 @@ export default function RemedyDetailsPage({ remedies }) {
     return <p>...loading</p>;
   }
 
+  function handleDelete(id) {
+    handleDeleteRemedy(id);
+    router.push("/");
+  }
```
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply pages/remedy/[id].js:20]: Insert consideration : 

```
 function deleteRemedyAndRedirect(id) {
    handleDeleteRemedy(id);
    router.push("/");
  }

```

File: components/ModalDelete.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+  padding: 14px 20px;
+  margin: 8px 0;
+  border: none;
+  cursor: pointer;
+  width: 20%;
+  opacity: 0.9;
+
+  &:hover {
+    opacity: 1;
+  }
```
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:55]: Change .close-button 

From : 
```
const CloseButton = styled.span`
  position: absolute;
  right: 15px;
  top: 10px;
  font-size: 40px;
  font-weight: bold;
  color: #f8fbca;
  cursor: pointer;

  &:hover {
    color: #f44336;
  }
`;
```

To

[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:55]: Change close button 

From : 

```
const DeleteButton = styled.button`
  background-color: #54582f;
  color: #f8fbca;
  padding: 14px 20px;
  margin: 8px 0;
  border: none;
  cursor: pointer;
  width: 20%;
  opacity: 0.9;

  &:hover {
    opacity: 1;
  }
`;
```

To :

File: components/ModalDelete.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:26]: Delete close-button 

```
const CloseButton = styled.span`
  position: absolute;
  right: 15px;
  top: 10px;
  font-size: 40px;
  font-weight: bold;
  color: #f8fbca;
  cursor: pointer;

  &:hover {
    color: #f44336;
  }
`;
```

File: components/ModalDelete.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+  justify-content: space-between;
+`;
+
+const CancelButton = styled.button`
+  background-color: #86895d;
+  color: #000;
+  padding: 14px;
+  margin: 8px 0;
+  border: none;
+  cursor: pointer;
```
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:112]: deletion of L.112

File: components/ModalDelete.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:55]:
```
.close-button 
....
....
opacity: 0.8;
transition: opacity 0.3s ease, transform 0.2s ease;

.close-button:active,
.close-button:focus {
  opacity: 1;
  transform: scale(1.1);
}
```
The :active and :focus states provide visual feedback when the button is tapped or focused on. This includes setting opacity: 1 and slightly scaling up the button using transform: scale(1.1) to give a tactile feel.The transition property ensures that the changes in opacity and scale happen smoothly, providing a better user experience.

File: components/ModalDelete.js [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+const Container = styled.div`
+  padding: 16px;
+  text-align: center;
+`;
+
+const CloseButton = styled.span`
+  position: absolute;
+  right: 15px;
+  top: 10px;
+  font-size: 40px;
```
[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:91]: Suggestion : 
A CSS variable `theme `can be defined to ensure consistency across the application. The variable can be used for font sizes, font weights, colors, padding, margins, and button width. 

```
const theme = {
  fontSizeLarge: '2.5rem', // Equivalent to 40px if base font size is 16px
  fontSizeMedium: '1rem', // Equivalent to 16px
  fontWeightBold: '700',
  colorPrimary: '#f8fbca', // Background color for modal content
  colorAccent: '#54582f',  // Border color
  colorHover: '#f44336',   // Example hover color for CloseButton
  colorButtonBackground: '#86895d', // Background color for CancelButton
  colorButtonConfirm: '#bec092',    // Background color for ConfirmDeleteButton
  colorButtonText: '#54582f',       // Text color for buttons
  paddingSmall: '1rem', // Equivalent to 16px
  paddingMedium: '0.875rem', // Equivalent to 14px
  marginSmall: '0.5rem', // Equivalent to 8px
  widthButton: '48%',
};
```

**Relative Units** 
We can replace the pixel values with `rem` to improve scalability and responsiveness. For example, 40px is converted to 2.5rem, assuming a base font size of 16px.
```

const Modal = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  z-index: 1;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(84, 88, 47, 0.9); // Same as colorAccent
`;

const ModalContent = styled.div`
  background-color: ${theme.colorPrimary}; // #f8fbca
  margin: auto;
  border: 1px solid ${theme.colorAccent}; // #54582f
  width: 80%;
  max-width: 31.25rem; /* Equivalent to 500px if base font size is 16px */
  padding: ${theme.paddingSmall};
  position: relative;
  text-align: center;
`;

const Container = styled.div`
  padding: ${theme.paddingSmall};
  text-align: center;
`;

const CloseButton = styled.span`
  position: absolute;
  right: 0.9375rem; /* Equivalent to 15px if base font size is 16px */
  top: 0.625rem; /* Equivalent to 10px if base font size is 16px */
  font-size: ${theme.fontSizeLarge};
  font-weight: ${theme.fontWeightBold};
  color: ${theme.colorPrimary}; // #f8fbca
  cursor: pointer;

  &:hover {
    color: ${theme.colorHover}; // Color for hover effect
  }
`;

const ButtonGroup = styled.div`
  display: flex;
  justify-content: space-between;
`;

const CancelButton = styled.button`
  background-color: ${theme.colorButtonBackground}; // #86895d
  color: ${theme.colorButtonText}; // #54582f
  padding: ${theme.paddingMedium};
  margin: ${theme.marginSmall} 0;
  border: none;
  cursor: pointer;
  width: ${theme.widthButton};

  &:hover {
    opacity: 0.8;
  }
`;

const ConfirmDeleteButton = styled.button`
  background-color: ${theme.colorButtonConfirm}; // #bec092
  color: ${theme.colorButtonText}; // #54582f
  padding: ${theme.paddingMedium};
  margin: ${theme.marginSmall} 0;
  border: none;
  cursor: pointer;
  width: ${theme.widthButton};

  &:hover {
    opacity: 0.8;
  }
`;
```

[the remark this answers is not in the document]
Comment (SylvieDoumet) [reply components/ModalDelete.js:91]: The 
&:hover {
    opacity: 0.8;
  }

Is just an add on that could be left in for desktop applications - for the version mobile first I can be left out

File: components/ModalDelete.js [hunk 10]
```diff
@@ -4,36 +4,26 @@ import styled from "styled-components";
...
     setShowModal(false);
     onDelete();
-  };
+  }
 
   return (
     <>
-      <DeleteButton onClick={handleDeleteClick}>Delete Remedy</DeleteButton>
-
+      <DeleteButton onClick={() => toggleModal()}>Delete Remedy</DeleteButton>
```
Comment (JessicaLoers) [inline components/ModalDelete.js:20]: You don't need to explicitly use a callback function if you pass the function as a reference

```suggestion
      <DeleteButton onClick={toggleModal}>Delete Remedy</DeleteButton>
```

File: components/ModalDelete.js [hunk 11]
```diff
@@ -4,36 +4,26 @@ import styled from "styled-components";
...
-    setShowModal(true);
-  };
-  const handleCloseModal = () => {
-    setShowModal(false);
-  };
+  function toggleModal() {
+    setShowModal(function (prevShowModal) {
+      return !prevShowModal;
+    });
+  }
```
Comment (JessicaLoers) [inline components/ModalDelete.js:11]: I wanted to mention that while using the `function` keyword in the `toggleModal` function is not incorrect, the following approach is more common and aligns with what we learned in the bootcamp.

⚠️ Unless you can provide us with a super valid reason for using the `function` keyword, please make sure to adhere to the conventions outlined in our curriculum.

```suggestion
 function toggleModal() {
  setShowModal((prevShowModal) => !prevShowModal);
}
```

File: components/ModalDelete.js [hunk 12]
```diff
@@ -4,36 +4,26 @@ import styled from "styled-components";
...
-                <ConfirmDeleteButton onClick={handleConfirmDelete}>
-                  Delete
-                </ConfirmDeleteButton>{" "}
-              </ButtonGroup>
-            </Container>
+        <Modal>
+          <ModalContent>
+            <p>Are you sure you want to delete the remedy?</p>
+            <button onClick={() => handleConfirmDelete()}>Delete</button>
+            <button onClick={() => toggleModal()}>Cancel</button>
```
Comment (JessicaLoers) [inline components/ModalDelete.js:26]: Same as above: You don't need to explicitly use a callback function if you pass the function as a reference

**Review by JessicaLoers — APPROVED** [review]
**Hi Team ✨**

I've noticed another pattern in the code that don't align with the learned practices and conventions from the Bootcamp curriculum. 

Please ensure that you stick to the writing styles and conventions we've been taught. Exceptions are always possible, but they should be justified accordingly.

Given that the Public Sprint Review is just around the corner and I completely understand that you'd like to present this feature, I will approve the PR. 

**_However, I trust that you'll incorporate my feedback accordingly._** 😇

File: components/ModalDelete.js [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+
+export default function RemedyDeleteButton({ onDelete }) {
+  const [showModal, setShowModal] = useState(false);
+
+  const handleDeleteClick = () => {
+    setShowModal(true);
+  };
+  const handleCloseModal = () => {
+    setShowModal(false);
+  };
```
[the remark this answers is not in the document]
Comment (JessicaLoers) [reply components/ModalDelete.js]: @SylvieDoumet 

There’s no need to add the changes here as comments — if you make these suggestions in the codebase, they will be visible in the updated PR.  

Also, I’m not able to review code that isn’t highlighted anyway.  😁

**Comment by Rahaf91** [comment]
Thank you, Jessica, for taking the time to review the code. I have updated it now. :)

**Pull request merged**